src/awkward/_v2/_connect/cuda/cuda_kernels/awkward_ListArray_validity.cu

```diff
@@ -0,0 +1,34 @@
+// BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+enum Error {
+    ERROR_START_STOP,    // message: "start[i] > stop[i]"
+    ERROR_START_ZERO,    // message: "start[i] < 0"
+    ERROR_STOP_CONTENT   // message: "stop[i] > len(content)"
+};
+
+template <typename C>
+__global__ void
+cuda_ListArray_validity(
+  const C* starts,
+  const C* stops,
+  int64_t length,
+  int64_t lencontent,
+  int64_t invocation_index,
+  int64_t* err_code) {
+  int64_t thread_id = blockIdx.x * blockDim.x + threadIdx.x;
+  if (thread_id < length && err_code[0] == MAX_NUMPY_INT) {
+    C start = starts[thread_id];
+    C stop = stops[thread_id];
+    if (start != stop) {
+      if (start > stop) {
+        atomicMin(err_code, invocation_index * (1 << ERROR_BITS) + ERROR_START_STOP); // failure("start[i] > stop[i]", i, kSliceNone, FILENAME(__LINE__));
+      }
+      if (start < 0) {
+        atomicMin(err_code, invocation_index * (1 << ERROR_BITS) + ERROR_START_ZERO); // failure("start[i] < 0", i, kSliceNone, FILENAME(__LINE__));
+      }
+      if (stop > lencontent) {
+        atomicMin(err_code, invocation_index * (1 << ERROR_BITS) + ERROR_STOP_CONTENT); // failure("stop[i] > len(content)", i, kSliceNone, FILENAME(__LINE__));
```

Comment:
The commented-out parts can be removed now.

Since this error-setting operation is one that would be done all over the place, maybe you want a utility function for it? Maybe even a [preprocessor macro](https://gcc.gnu.org/onlinedocs/cpp/Function-like-Macros.html#Function-like-Macros)? I don't know what `cupy.RawKernel` lets you do. Ideally, you'd like to be able to write these lines as

```c++
raise_error(ERROR_STOP_CONTENT)
```

and the `raise_error` macro would define once-and-for-all the `atomicMin`, the fact that we're overwriting `err_code`, the arithmetic on `invocation_index` and `ERROR_BITS`, etc. If it can't be a preprocessor macro but can be a function, you'd have to pass `err_code` and `invocation_index` as arguments, along with the specific error.

Reply:
Turned this into a `preprocessor macro` `RAISE_ERROR(ERROR)`